File: src/test/java/fr/eulbobo/dojo/byron/domain/DisasterMuseTest.java
package fr.eulbobo.dojo.byron.domain;

import org.junit.jupiter.api.Test;
import org.mockito.Mockito;

import static org.mockito.Mockito.*;

class DisasterMuseTest {

    @Test
    void shouldGetDisasterMuseAfter3rollsResultingIn6() {
        Sanity sanity = Mockito.mock(Sanity.class);
        EventLog eventLog = Mockito.mock(EventLog.class);
        DisasterMuse disasterMuse = new DisasterMuse(sanity, eventLog);
        disasterMuse.diceRollIs(DiceRoll.of(6));
        verifyNoInteractions(sanity, eventLog);
        disasterMuse.diceRollIs(DiceRoll.of(6));
        verifyNoInteractions(sanity, eventLog);
        disasterMuse.diceRollIs(DiceRoll.of(6));
        verify(sanity).disaster();
        verify(eventLog).log("Byron destroys your manuscript either by accident or on purpose, during one of his episodes");
        verifyNoMoreInteractions(sanity);
    }

    @Test
    void shouldNotGetDisasterMuseAfterAFourth6Roll() {
        Sanity sanity = Mockito.mock(Sanity.class);
        EventLog eventLog = Mockito.mock(EventLog.class);
        DisasterMuse disasterMuse = new DisasterMuse(sanity, eventLog);
        disasterMuse.diceRollIs(DiceRoll.of(6));
        disasterMuse.diceRollIs(DiceRoll.of(6));
        disasterMuse.diceRollIs(DiceRoll.of(6));
        disasterMuse.diceRollIs(DiceRoll.of(6));
        verify(sanity).disaster();
        verify(eventLog).log("Byron destroys your manuscript either by accident or on purpose, during one of his episodes");
        verifyNoMoreInteractions(sanity, eventLog);
    }
}

File: c#/fr.eulbobo.dojo.byron/fr.eulbobo.dojo.byron.tests/DisasterMuseTest.cs
using fr.eulbobo.dojo.byron.domain;
using Moq;

namespace fr.eulbobo.dojo.byron.tests
{
    public class DisasterMuseTest
    {
        [Test]
        public void ShouldGetDisasterMuseAfter3rollsResultingIn6()
        {
            var sanity = new Mock<Sanity>();
            var eventLog = new Mock<EventLog>();
            DisasterMuse disasterMuse = new DisasterMuse(sanity.Object, eventLog.Object);
            disasterMuse.DiceRollIs(DiceRoll.Of(6));
            sanity.VerifyNoOtherCalls();
            eventLog.VerifyNoOtherCalls();
            disasterMuse.DiceRollIs(DiceRoll.Of(6));
            sanity.VerifyNoOtherCalls();
            eventLog.VerifyNoOtherCalls();
            disasterMuse.DiceRollIs(DiceRoll.Of(6));
            sanity.Verify(s => s.Disaster());
            eventLog.Verify(evtl => evtl.Log("Byron destroys your manuscript either by accident or on purpose, during one of his episodes"));
            sanity.VerifyNoOtherCalls();
        }

        [Test]
        public void ShouldNotGetDisasterMuseAfterAFourth6Roll()
        {
            var sanity = new Mock<Sanity>();
            var eventLog = new Mock<EventLog>();
            DisasterMuse disasterMuse = new DisasterMuse(sanity.Object, eventLog.Object);
            disasterMuse.DiceRollIs(DiceRoll.Of(6));
            disasterMuse.DiceRollIs(DiceRoll.Of(6));
            disasterMuse.DiceRollIs(DiceRoll.Of(6));
            disasterMuse.DiceRollIs(DiceRoll.Of(6));
            sanity.Verify(s => s.Disaster());
            eventLog.Verify(evtl => evtl.Log("Byron destroys your manuscript either by accident or on purpose, during one of his episodes"));
            sanity.VerifyNoOtherCalls();
            eventLog.VerifyNoOtherCalls();
        }
    }
}
